packages/docusaurus/src/client/preload.ts

```diff
@@ -17,7 +17,9 @@ import {matchRoutes} from 'react-router-config';
  * @returns Promise object represents whether pathname has been preloaded
  */
 export default function preload(pathname: string): Promise<void[]> {
-  const matches = matchRoutes(routes, pathname);
+  const matches = [pathname, decodeURI(pathname)]
+    .map(p => matchRoutes(routes, p))
+    .flat();
```

Comment:
agree preload is cached and won't run twice

Now I'd rather still dedup the string because if the string is not encoded we would run the matching twice. Dedup would ensure we only run it once in most cases.

You can use `Array.from(new Set(arr))` to dedup

```suggestion
  const matches = Array.from(new Set([pathname, decodeURI(pathname)]))
    .map(p => matchRoutes(routes, p))
    .flat();
```